sources/core/Xenko.Core.Mathematics/Color4.cs

```diff
@@ -159,13 +159,13 @@ public Color4(float[] values)
         {
             if (values == null)
                 throw new ArgumentNullException(nameof(values));
-            if (values.Length != 4)
-                throw new ArgumentOutOfRangeException(nameof(values), "There must be four and only four input values for Color4.");
+            if (values.Length < 3)
+                throw new ArgumentOutOfRangeException(nameof(values), "There must be at least 3 float[] values for Color4.");
```

Comment:
Seems like a bad idea to allow non-3 non-4 size arrays.

Reply:
Made the change. My original thought was, there was enough values for the function to work, so I'd let the function work.